skimage/_shared/fast_exp.h

```diff
@@ -38,3 +41,64 @@ __inline double fast_exp (double y)
 
     return _eco.d;
 }
+
+/* Simple cast to float */
+__inline float _fast_expf(float y)
```

Comment:
I don't see that this version is still being used anywhere.  Isn't `fast_expf` now used instead?

Reply:
Hi grlee77, thank you very much for your review, I am currently doing some tests, and I am observing that the `fast_expf` implementation suffers a lot from the stair case effect described in the reference paper and I am hesitating to simply using this version `_fast_expf` with a simple cast from single to double precision which is the only overhead compared to the double precision implementation and still faster then the `expf` function.
I will post some figures to illustrate this, like @blochl did in #2878.

Reply:
Here is the figure
![fast_exp](https://user-images.githubusercontent.com/3438227/70043035-2b257280-15c0-11ea-9af8-44f86901a6b9.png)
As you can see, not really good :-(
May be @blochl, if you still have the data shown in your previous [PR](https://github.com/scikit-image/scikit-image/pull/2878#issuecomment-346344773), can you have a try with this implementation and see if you observe the same artifacts? Thank you.

Reply:
The inaccuracy shouldn't matter much for this application, but I would be fine with using the typecast version instead. I doubt it makes a substantial difference in run time either way.

Reply:
> May be @blochl, if you still have the data shown in your previous PR

Sorry, I don't have it anymore.
___
...And I still don't get why `{_,}fast_expf` are needed (reasoning in my review comment).

Reply:
As @grlee77, I would not worry too much here about these (small) artifacts. The weights used in the non-local means algorithm are quite ad hoc anyway.